comdb2/cdb2.py

```diff
@@ -396,7 +396,14 @@ def execute(
         """
         if parameters is None:
             parameters = {}
-        self._cursor = iter(self._hndl.execute(sql, parameters))
+
+        column_types_array: Sequence[int] | None = None
+        if column_types is not None:
+            column_types_array = array.array("i", column_types)
+        if not column_types_array:
+            column_types_array = None
```

Comment:
If I understand this correctly, this means that an empty `column_types` argument will have the same effect as specifying `None` or not specifying anything. Is it what we want? After reading the documentation, I would have expected an error to be raised because "the number of elements in ``column_types`` doesn't match the number of columns in the result set"

Reply:
I'd also have expected an error to be raised, but Comdb2 doesn't, and we can't reasonably validate this ourselves. We can file a bug report with them about it, though.

Reply:
The Comdb2 folks have confirmed that's intended behavior. I've updated the docstring to explicitly say that it's ignored if empty, and added tests to prove it.